File: Android/ImageResize-src/ImageResize.java
package com.magicsoftware.magicdev;

import android.graphics.Matrix;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import java.io.ByteArrayOutputStream;
import java.io.FileOutputStream;
import java.io.File;
import java.io.IOException;
import android.content.Context;
import com.magicsoftware.core.CoreApplication;

public class ImageResize {
	public static String resize(String imagePath,int compressionQuality,int Width,int Height) {
		// syntax: resize:ImagePath,compressionQuality,Width,Height
		// returns: the new resized image file path
		// in compressionQuality - 0 being the maximum compression, 100 being the best quality
		// in Width and Height, value of 0 means keeping the original size
	 
		// load the image
		Bitmap image = BitmapFactory.decodeFile(imagePath);
		// resize the image
		if(Width == 0)
			Width=image.getWidth();
		if(Height == 0)
			Height=image.getHeight();
		Bitmap resizedImage = Bitmap.createScaledBitmap(image, Width, Height, true);
		// compress the image
		ByteArrayOutputStream blob = new ByteArrayOutputStream();
		resizedImage.compress(Bitmap.CompressFormat.JPEG, compressionQuality, blob);
		// resizedImage.compress(Bitmap.CompressFormat.PNG, 0, blob); //compression quality is ignored for png 
		// save the image to a temp file in the iOS temp folder
		File tempFile = new File(CoreApplication.getInstance().getApplicationContext().getCacheDir(),"tempfile");
		try {
			FileOutputStream fos = new FileOutputStream (tempFile); 
			blob.writeTo(fos);
			fos.close();
		}
		catch(IOException ioe) {
			System.out.println("IOException : " + ioe);
		}
		return tempFile.getAbsolutePath();
	}
	
	public static String rotate(String imagePath,int angel) {
		// syntax: rotate:ImagePath,angel (can also be negative)
		// returns: the new rotated image file path
	 
		// load the image
		Bitmap image = BitmapFactory.decodeFile(imagePath);
        // create a matrix object
        Matrix matrix = new Matrix();
        matrix.postRotate(angel); 
        // create a new bitmap from the original using the matrix to transform the result
        Bitmap rotatedImage = Bitmap.createBitmap(image , 0, 0, image.getWidth(), image.getHeight(), matrix, true);
		ByteArrayOutputStream blob = new ByteArrayOutputStream();
		rotatedImage.compress(Bitmap.CompressFormat.PNG, 0, blob); 
		// save the image to a temp file in the iOS temp folder
		File tempFile = new File(CoreApplication.getInstance().getApplicationContext().getCacheDir(),"tempfile");
		try {
			FileOutputStream fos = new FileOutputStream (tempFile); 
			blob.writeTo(fos);
			fos.close();
		}
		catch(IOException ioe) {
			System.out.println("IOException : " + ioe);
		}
		return tempFile.getAbsolutePath();
	}
}

File: iOS/ImageResize-MagicApp/ImageResize.m
//
//  ImageResize.m
//  MagicApp
//

#import "ImageResize.h"

@interface ImageResize ()

@end

@implementation ImageResize

+(NSString *)resize:(NSArray *)params
{
	// syntax: resize:ImagePath,compressionQuality,Width,Height
	// returns: the new resized image file path
	// in compressionQuality - 0 being the maximum compression, 100 being the best quality
	// in Width and Height, value of 0 means keeping the original size
	
	NSString *imagePath = [params objectAtIndex: 0];
	CGFloat compressionQuality = [[params objectAtIndex: 1] floatValue]; 
	float Width = [[params objectAtIndex: 2] floatValue];
	float Height = [[params objectAtIndex: 3] floatValue];
	
	// load the image
	NSData *imgdata = [NSData dataWithContentsOfFile:imagePath]; 
	UIImage *image = [UIImage imageWithData:imgdata];
	// resize the image
	if(Width == 0)
		Width=image.size.width;
	if(Height == 0)
		Height=image.size.height;
	CGRect rect = CGRectMake(0.0, 0.0, Width, Height);
	UIGraphicsBeginImageContext(rect.size);
	[image drawInRect:rect];
	UIImage *img = UIGraphicsGetImageFromCurrentImageContext();
	UIGraphicsEndImageContext();
	// compress the image
	NSData *imageData = [NSData dataWithData:UIImageJPEGRepresentation(img, compressionQuality/100)];
	// save the image to a temp file in the iOS temp folder
	NSString * tempFile = [NSString pathWithComponents:@[NSTemporaryDirectory(), @"tempfile"]];
	[imageData writeToFile:tempFile atomically:YES];                                             
	return tempFile;
}

+(NSString *)rotate:(NSArray *)params
{
	// syntax: rotate:ImagePath,angel (can also be negative)
	// returns: the new rotated image file path

	NSString *imagePath = [params objectAtIndex: 0];
	CGFloat angel = [[params objectAtIndex: 1] floatValue]; 
	
	// load the image
	NSData *imgdata = [NSData dataWithContentsOfFile:imagePath]; 
	UIImage *image = [UIImage imageWithData:imgdata];
	// Calculate the size of the rotated view's containing box for our drawing space
	UIView *rotatedViewBox = [[UIView alloc] initWithFrame:CGRectMake(0,0,image.size.width, image.size.height)];
	CGAffineTransform t = CGAffineTransformMakeRotation(angel * M_PI / 180);
	rotatedViewBox.transform = t;
	CGSize rotatedSize = rotatedViewBox.frame.size;
	// Create the bitmap context
	UIGraphicsBeginImageContext(rotatedSize);
	CGContextRef bitmap = UIGraphicsGetCurrentContext();
	// Move the origin to the middle of the image so we will rotate and scale around the center.
	CGContextTranslateCTM(bitmap, rotatedSize.width/2, rotatedSize.height/2);
	// Rotate the image context
	CGContextRotateCTM(bitmap, (angel * M_PI / 180));
	//Draw the rotated/scaled image into the context
	CGContextScaleCTM(bitmap, 1.0, -1.0);
	CGContextDrawImage(bitmap, CGRectMake(-image.size.width / 2, -image.size.height / 2, image.size.width, image.size.height), [image CGImage]);
	UIImage *img = UIGraphicsGetImageFromCurrentImageContext();
	UIGraphicsEndImageContext();
	NSData *imageData = [NSData dataWithData:UIImageJPEGRepresentation(img, 1)];
	// save the image to a temp file in the iOS temp folder
	NSString * tempFile = [NSString pathWithComponents:@[NSTemporaryDirectory(), @"tempfile"]];
	[imageData writeToFile:tempFile atomically:YES];                                             
	return tempFile;
}

@end
